File: Proxy/c#/Proxy/MyCustomDnsProxy.cs
using System.Collections.Generic;
using System.Linq;

namespace Proxy
{
    public class MyCustomDnsProxy
    {
        private readonly WorldDns _worldDns;

        public MyCustomDnsProxy(WorldDns worldDns)
        {
            _worldDns = worldDns;
        }

        private List<string> BlackList = new List<string>
        {
            {"example.com"}
        };

        public string GetAddress(string domain)
        {
            if (BlackList.Any(_ => _ == domain))
            {
                return "Forbidden Access is Denied!";
            }
            else
            {
                return _worldDns.GetAddress(domain);
            }
        }
    }
}

File: Proxy/python/mycustom_dns_proxy.py

from worlddns import WorldDns

class MyCustomDnsProxy(): ## The proxy class that wraps original dns class
    def __init__(self, world_dns:WorldDns):
        self.world_dns= world_dns
        self.black_list = {"example.com","sample.com"}
    
    def get_address(self,domain:str):
        if domain not in self.black_list:
            return self.world_dns.get_address(domain)
        else:
            return "Forbbiden Access is Denied!"
